recipes/paho-mqtt-cpp/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class PahoMqttCppConan(ConanFile):
+    name = "paho-mqtt-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/eclipse/paho.mqtt.cpp"
+    topics = ("MQTT", "IoT", "eclipse", "SSL", "paho", "Cpp")
+    license = "EPL-1.0"
+    description = """The open-source client implementations of MQTT and MQTT-SN"""
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "ssl": [True, False],
+               }
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "ssl": False
+                       }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Paho cpp can not be built as shared on Windows.")
+        self.options["paho-mqtt-c"].shared = self.options.shared
+        self.options["paho-mqtt-c"].ssl = self.options.ssl
+
+
+    def requirements(self):
+        self.requires("paho-mqtt-c/1.3.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
```

Comment:
configure() is duplicated ... I spent 20 min to see it :man_facepalming:

Reply:
something we can check in hooks?

Reply:
Yes, but I can say pylint is able to warn about it, and we already have pylint hook, but it's not enabled.